apiclient/client.go

```diff
@@ -144,6 +146,11 @@ func (c *HTTPclient) SetRetries(n int) {
 	c.retries = n
 }
 
+func (c *HTTPclient) SetTimeout(d time.Duration) {
+	c.c.Timeout = d
+	c.c.Transport.(*http.Transport).ResponseHeaderTimeout = d
```

Comment:
i have the feeling this can panic. better make an assertion and simply skip if it's not http.Transport

Reply:
```go
func (c *HTTPclient) SetTimeout(d time.Duration) {
	c.c.Timeout = d
	if transport, ok := c.c.Transport.(*http.Transport); ok {
		transport.ResponseHeaderTimeout = d
	}
}

```

Reply:
i mean in our codebase it's always http.Transport. but it also doesn't hurt to avoid panics